streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamMap.java

```diff
@@ -42,6 +44,7 @@ public KStreamMap(final KeyValueMapper<? super KIn, ? super VIn, ? extends KeyVa
         public void process(final Record<KIn, VIn> record) {
             final KeyValue<? extends KOut, ? extends VOut> newPair =
                 mapper.apply(record.key(), record.value());
+            Objects.requireNonNull(newPair, String.format("KeyValueMapper can't return null from mapping the record: %s", record));
```

Comment:
Just a suggestion:
```suggestion
            Objects.requireNonNull(newPair, "The provided KeyValueMapper returned null which is not allowed.");
```

BTW, we should not output records since they might contain sensitive data.